### PR TITLE
Added functionality to plot rho for a specific species

### PR DESCRIPTION
**Pull request by EtherealMC-Bit**
Utilities the ability to gather rho for a specific species in order to allow diagnostic plotting of rho for a specific species.

### CODE REVIEW

File: mewarpx/mewarpx/diags_store/field_diagnostic.py [hunk 1]
```diff
@@ -47,6 +49,7 @@ def __init__(self, diag_steps, diag_data_list, grid, name, write_dir,
         self.write_dir = write_dir
         self.plot_on_diag_step = plot_on_diag_step
         self.plot_data_list = plot_data_list
+        self.plot_species_list = plot_species_list
```
Comment (roelof-groenewald) [inline mewarpx/mewarpx/diags_store/field_diagnostic.py]: Could you please make this behave as follows (suggestion is untested so please confirm that it works)
```suggestion
        if plot_species_list is not None:
            self.plot_species_list = plot_species_list
        else:
            self.plot_species_list = [species.name for species in mwxrun.species]
```

**Review by roelof-groenewald — CHANGES_REQUESTED** [review]
Just one change suggestion. Please let me know if the intended change is not clear.

**Review by roelof-groenewald — APPROVED** [review]
Thanks!

**Review by KZhu-ME — APPROVED** [review]
Overall, looks good!

**Pull request merged**